CHANGELOG.md

```diff
@@ -1,6 +1,20 @@
 CHANGELOG
 =========
 
+### 4.0.0 (2016-xx-xx)
+
+* Added: Interface for `Geocoder\Model\AddressCollection` called `Geocoder\GeocoderResult`. Public APIs are updated to type hint for `Geocoder\GeocoderResult`.
```

Comment:
> but I still question the need for an interface here

An interface wont hurt and it opens up the possibility for others custom providers to return a `CollectionOnSteroids` and still comply with the contract.